cmd/ddev/cmd/root.go

```diff
@@ -194,6 +195,149 @@ func init() {
+								Name:        subCmd.Name(),
+								Description: subCmd.Short,
+							})
+						}
+					}
+					jsonCommandGroups = append(jsonCommandGroups, &jsonCommandGroup{
+						Name:     group.Title,
+						Commands: jsonGroupCommands,
+					})
+				}
+				// Subcommands that don't belong in a group
+				if !command.AllChildCommandsHaveGroup() {
+					for _, subCmd := range command.Commands() {
+						if subCmd.GroupID == "" && (subCmd.IsAvailableCommand() || subCmd.Name() == "help") {
+							jsonAdditionalCommands = append(jsonAdditionalCommands, &jsonSubCommand{
+								Name:        subCmd.Name(),
+								Description: subCmd.Short,
+							})
+						}
+					}
+				}
+			}
+
+			// Additional help commands
+			for _, subCmd := range command.Commands() {
+				if subCmd.IsAdditionalHelpTopicCommand() {
+					jsonAdditionalHelpCommands = append(jsonAdditionalHelpCommands, &jsonSubCommand{
+						Name:        subCmd.Name(),
+						Description: subCmd.Short,
+					})
+				}
+			}
+
+			// Build list of all non-global flags
+			command.LocalFlags().VisitAll(func(localFlag *pflag.Flag) {
+				jsonFlags = append(jsonFlags, &jsonFlag{
+					Name:                localFlag.Name,
+					Shorthand:           localFlag.Shorthand,
+					Usage:               localFlag.Usage,
+					Default:             localFlag.DefValue,
+					Deprecated:          localFlag.Deprecated,
+					Hidden:              localFlag.Hidden,
+					ShorthandDeprecated: localFlag.ShorthandDeprecated,
+				})
+			})
+
+			// Build list of all global flags
+			command.InheritedFlags().VisitAll(func(globalFlag *pflag.Flag) {
+				jsonGlobalFlags = append(jsonGlobalFlags, &jsonFlag{
+					Name:                globalFlag.Name,
+					Shorthand:           globalFlag.Shorthand,
+					Usage:               globalFlag.Usage,
+					Default:             globalFlag.DefValue,
+					Deprecated:          globalFlag.Deprecated,
+					Hidden:              globalFlag.Hidden,
+					ShorthandDeprecated: globalFlag.ShorthandDeprecated,
+				})
+			})
+
+			// Build usage, reflecting what is rendered for the plain text output
+			var fullUsage = command.UseLine()
+			if command.HasAvailableSubCommands() {
+				fullUsage = fullUsage + "\n" + command.CommandPath() + " [command]"
+			}
+
+			rawResult := make(map[string]interface{})
+			rawResult["ShortDescription"] = command.Short
+			rawResult["LongDescription"] = command.Long
+			rawResult["Usage"] = fullUsage
+			rawResult["Aliases"] = command.Aliases
+			rawResult["Example"] = command.Example
+			rawResult["Commands"] = jsonCommands
+			rawResult["CommandGroups"] = jsonCommandGroups
+			rawResult["AdditionalCommands"] = jsonAdditionalCommands
+			rawResult["AdditionalHelpCommands"] = jsonAdditionalHelpCommands
+			rawResult["Flags"] = jsonFlags
+			rawResult["GlobalFlags"] = jsonGlobalFlags
+			rawResult["Deprecated"] = command.Deprecated
+			rawResult["Hidden"] = command.Hidden
+
+			output.UserOut.WithField("raw", rawResult).Print()
```

Comment:
Note that I can't pass the raw help output into the `Print()` call, since the original help function prints directly to the terminal rather than returning its result.